### PR TITLE
Snack Bar implemented throughout the app

### LINKED ISSUE #71: Replace Toasts with bottom sheet that appears for 3 secs

### PR DESCRIPTION
**Pull request by Ammar-Ishfaq**
Resolves #71

### CODE REVIEW

File: app/src/main/java/com/waseefakhtar/doseapp/DoseApp.kt [hunk 1]
```diff
@@ -58,6 +65,9 @@ fun DoseApp() {
             modifier = Modifier.fillMaxSize(),
             color = MaterialTheme.colorScheme.background
         ) {
+            // For the snackbar
+            var openMySnackbar by remember { mutableStateOf(false) }
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/DoseApp.kt:69]: Let's follow the convention here: 

```suggestion
            var snackbarVisibility by remember { mutableStateOf(false) }
```

File: app/src/main/java/com/waseefakhtar/doseapp/DoseApp.kt [hunk 2]
```diff
@@ -58,6 +65,9 @@ fun DoseApp() {
...
+            var snackBarMessage by remember { mutableStateOf("") }
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/DoseApp.kt:70]: I'd also like the naming to be similar to Material Design's naming:

```suggestion
            var snackbarMessage by remember { mutableStateOf("") }
```

So we can replace `snackBar` with `snackbar` in all other places too. :)

File: app/src/main/java/com/waseefakhtar/doseapp/util/SnackBarUtil.kt [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+import kotlinx.coroutines.launch
+
+class SnackBarUtil {
+
+    companion object {
+        @Composable
+        fun SnackbarWithoutScaffold(
+            message: String,
+            showSb: Boolean,
+            openSnackbar: (Boolean) -> Unit
```
Comment (waseefakhtar) [inline app/src/main/java/com/waseefakhtar/doseapp/util/SnackBarUtil.kt:29]: Let's go with:

```suggestion
            isVisible: Boolean,
            onVisibilityChange: (Boolean) -> Unit
```

**Review by waseefakhtar — COMMENTED** [review]
Looks pretty good and works well! I just had some suggestions regarding naming conventions.

**Comment by Ammar-Ishfaq** [comment]
Certainly, I've made the changes you recommended by making commit suggestions. 😊

**Comment by Ammar-Ishfaq** [comment]
Hello @waseefakhtar, I've improved the implementation of the Snackbar feature. Now, you can simply call the function (showSnackbar()) without the need to create parameters within it. This eliminates any unnecessary boilerplate code, making it a more efficient and user-friendly experience for future developers. I'm eager to hear your thoughts and hope we can proceed with merging this improvement. 😊

**Review by waseefakhtar — APPROVED** [review]
Looks pretty good and nice improvement with eliminating unnecessary boilerplate code! I just made some minor changes renaming variables and functions and fixing ktlint. Thank you for your contribution, @Ammar-Ishfaq! :)

**Pull request merged**